buffer/src/test/java/io/netty/buffer/AbstractPooledByteBufTest.java

```diff
@@ -34,4 +38,26 @@ protected ByteBuf newBuffer(int length, int maxCapacity) {
         assertEquals(0, buffer.readerIndex());
         return buffer;
     }
+
+    @Test
+    public void ensureWritableWithEnoughSpaceShouldNotThrow() {
+        ByteBuf buf = buffer(1);
+        int originalWriterIndex = buf.writerIndex();
+        buf.writerIndex(buf.writerIndex() + 1);
+        buf.ensureWritable(1);
+        assertEquals(originalWriterIndex + 1, buf.writerIndex());
+        assertEquals(1, buf.readableBytes());
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void ensureWritableWithNotEnoughSpaceShouldThrow() {
+        ByteBuf buf = buffer(1);
+        try {
+            buf.writerIndex(buf.writerIndex() + 2);
```

Comment:
you want to use `buf.writerIndex(buf.writerIndex() + buf.writableBytes())` if I am not wrong.